core/src/main/scala/org/apache/spark/scheduler/StageInfo.scala

```diff
@@ -39,7 +39,9 @@ class StageInfo(
     val taskMetrics: TaskMetrics = null,
     private[spark] val taskLocalityPreferences: Seq[Seq[TaskLocation]] = Seq.empty,
     private[spark] val shuffleDepId: Option[Int] = None,
-    val resourceProfileId: Int) {
+    val resourceProfileId: Int,
+    private[spark] var isPushBasedShuffleEnabled: Boolean = false,
+    private[spark] var shuffleMergerCount: Int = 0) {
```

Comment:
Why do we need these ?

Reply:
But they are `private[spark]` with no consumers within spark right now, no ?

Reply:
Yes, currently there are no consumers with in Spark. But [[SPARK-36620][SHUFFLE] Add client side push based shuffle metrics](https://github.com/apache/spark/pull/34000) has to use these metrics and populate it which is currently ongoing. Do you think we can move this change as well to that PR or wait for this one to get merged and then correspondingly make the changes in the other PR to consume these metrics? cc @thejdeep